swift-sdk/Internal/InAppInternal.swift

```diff
@@ -5,59 +5,42 @@
 
 import Foundation
 
-/// Callbacks from the synchronizer
-protocol InAppSynchronizerDelegate : class {
-    func onInAppRemoved(messageId: String)
-    func onInAppMessagesAvailable(messages: [IterableInAppMessage])
-}
-
 ///
 protocol InAppSynchronizerProtocol {
-    // These variables are used for callbacks
-    var internalApi: IterableAPIInternal? {get set}
-    var inAppSyncDelegate: InAppSynchronizerDelegate? {get set}
-    
-    // These methods are called on new messages arrive etc.
-    func sync()
-    func remove(messageId: String)
+    // Fetch from server and sync
+    func sync() -> Future<[IterableInAppMessage]>
+}
+
+/// For callbacks when silent push notifications arrive
+protocol InAppNotifiable {
+    func onInAppSyncNeeded()
+    func onInAppRemoved(messageId: String)
 }
 
 extension IterableInAppTriggerType {
     static let defaultTriggerType = IterableInAppTriggerType.immediate // default is what is chosen by default
     static let undefinedTriggerType = IterableInAppTriggerType.never // undefined is what we select if payload has new trigger type
 }
 
-class InAppSilentPushSynchronizer : InAppSynchronizerProtocol {
-    weak var internalApi: IterableAPIInternal?
-    weak var inAppSyncDelegate: InAppSynchronizerDelegate?
-    
+class InAppSynchronizer : InAppSynchronizerProtocol {
     init() {
         ITBInfo()
     }
     
-    func sync() {
+    func sync() -> Future<[IterableInAppMessage]> {
         ITBInfo()
-        guard let internalApi = self.internalApi else {
+        guard let internalApi = IterableAPI.internalImplementation else {
             ITBError("Invalid state: expected InternalApi")
-            return
+            return Promise(error: IterableError.general(description: "Invalid state: expected InternalApi"))
         }
-        
-        InAppHelper.getInAppMessagesFromServer(internalApi: internalApi, number: numMessages).onSuccess {
-                self.inAppSyncDelegate?.onInAppMessagesAvailable(messages: $0)
-            }.onError {
-                ITBError($0.localizedDescription)
-        }
-    }
-    
-    func remove(messageId: String) {
-        ITBInfo()
-        inAppSyncDelegate?.onInAppRemoved(messageId: messageId)
+
+        return InAppHelper.getInAppMessagesFromServer(internalApi: internalApi, number: numMessages)
     }
-    
+
     deinit {
         ITBInfo()
     }
     
     // how many messages to fetch
-    private let numMessages = 10
+    private let numMessages = 100
```

Comment:
Do we want to change this in the current beta (6.1.0)?

Reply:
Yes we probably should. 
Later on, if needed, maybe we should think about having a config parameter.

Reply:
Yeah, 10 messages is low enough that some customers might hit that limit during implementation..